src/rubrix/client/sdk/commons/api.py

```diff
@@ -112,13 +112,16 @@ def build_data_response(
     response: httpx.Response, data_type: Type[T]
 ) -> Response[List[T]]:
     if 200 <= response.status_code < 400:
-        parsed_response = [data_type(**json.loads(r)) for r in response.iter_lines()]
-        return Response(
-            status_code=response.status_code,
-            content=b"",
-            headers=response.headers,
-            parsed=parsed_response,
-        )
+        try:
+            parsed_response = [data_type(**json.loads(r)) for r in response.iter_lines()]
+            return Response(
+                status_code=response.status_code,
+                content=b"",
+                headers=response.headers,
+                parsed=parsed_response,
+            )
+        except httpx.RemoteProtocolError as err:
+            raise Exception(f"Malformed query!") from None
```

Comment:
So, that's mean that all error of this kind received from the server will be considered as a malformed query, but we can have errors of different nature, right?

Maybe, we should work on an error mechanism exchange when streaming has already started. Let's me investigate and make you a design proposal, ok?

Reply:
Hi @frascuchon 
Thanks for the review.

>So, that's mean that all error of this kind received from the server will be considered as a malformed query, but we can have errors of different nature, right?

RemoteProtocolError comes mostly when protocol is violated by  the server for eg: when it"s send the malformed HTTP. Error of different nature are most likely to be captured via explicit non 2XX error codes being sent back from server. Hence I attributed this particular exception to "Malformed query" but definitely there can be some other cases which might lead to same exception.
 
I suppressed the exception(`raise Exception(f"Malformed query!") from None`) here because the original error trace was nested, too verbose and not actionable by the user.

> Maybe, we should work on an error mechanism exchange when streaming has already started. Let's me investigate and make you a design proposal, ok?

Design proposal would be very helpful. Looking forward to it.